content/chapters/template-chapter/lecture/slides.mdpp

```diff
@@ -11,1121 +11,42 @@ revealOptions:
-    * Otherwise, the new process takes over
-
-----
-
-### Create a new process - `fork` + `exec`
-
-* [Quiz](./quiz/exec-without-fork.md)
-* "Save" the shell before calling `exec("/bin/ls")`
-* Use `fork` and only the child calls `exec`
-
-![Fork and Exec Diagram](media/fork-exec.svg)
-
-----
-
-### Linux Syscalls
-
-```
-student@os:~/.../compute/lecture/demo/create-process$ strace ./fork_exec
-[...]
-clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f7e83aa4810) = 5279
-[...]
-```
-* `fork` and `pthread_create` are `libc` wrappers to the same syscall: `clone`
-
-----
-
-### Libc vs Linux vs Windows
-
-* Libc approach - `system`:
-
-``` [1 | 2 | 3 | 5 | 7 | 9]
-student@os:~/.../compute/lecture/demo/create-process$ strace -ff -e clone,execve ./system
-execve("./system", ["./system"], 0x7ffff3f20008 /* 60 vars */) = 0
-clone(child_stack=0x7f3fa624aff0, flags=CLONE_VM|CLONE_VFORK|SIGCHLDstrace: Process 117770 attached
- <unfinished ...>
-[pid 117770] execve("/bin/sh", ["sh", "-c", "ls -la"], 0x7ffdac7acfe8 /* 60 vars */
-[...]
-[pid 117770] clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f9292b28850) = 117771
-strace: Process 117771 attached
-[pid 117771] execve("/usr/bin/ls", ["ls", "-la"], 0x55c700084048 /* 60 vars */) = 0
-[...]
-```
-
-----
-
-### Libc vs Linux vs Windows
-
-* `system` first creates a new shell (`execve("/bin/sh"`)
-* This shell now spawns `ls -la` (`execve("/usr/bin/ls", ["ls", "-la"]`)
-
-----
-
-### Libc vs Linux vs Windows
-
-* Linux approach - `posix_spawn` and `fork` + `exec`:
-
-``` [1 | 2 | 3 | 5 | 7 | 8 | 9 | 11 | 13]
-student@os:~/.../compute/lecture/demo/create-process$ strace -ff -e clone,execve ./posix_spawn
-execve("./posix_spawn", ["./posix_spawn"], 0x7ffcfed53928 /* 60 vars */) = 0
-clone(child_stack=0x7f842691eff0, flags=CLONE_VM|CLONE_VFORK|SIGCHLDstrace: Process 118949 attached
- <unfinished ...>
-[pid 118949] execve("/bin/ls", ["-la"], 0x7ffc97832878 /* 60 vars */ <unfinished ...>
-
-student@os:~/.../compute/lecture/demo/create-process$ strace -ff -e clone,execve ./fork_exec
-execve("./fork_exec", ["./fork_exec"], 0x7ffdb2c85028 /* 60 vars */) = 0
-clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLDstrace: Process 119051 attached
-[...]
-[pid 119051] execve("/home/student/.local/bin/ls", ["-la"], 0x7ffee90fee78 /* 60 vars */) = -1
-[...]
-[pid 119051] execve("/usr/bin/ls", ["-la"], 0x7ffee90fee78 /* 60 vars */) = 0
-```
-
-* The "manual" method (`fork` + `exec`) tries all `$PATH` entries before finding `"/usr/bin/ls"`
-
-----
-
-### Libc vs Linux vs Windows
-
-* Windows approach: `CreateProcess`
-
-```
-PS C:\...\lecture\demo\create-process> .\create_process.exe
-[parent] PID = 11876; child PID = 10376; Waiting...
-[...]
-```
-
-* The command given to `CreateProcess` must specify the interpreter: `cmd`
-
----
-
-### `fork` vs `pthread_create`
+* You can even animate slides like this
```

Comment:
The "animation" below is but a single type. You can add transitions by using HTML tags: ``. Also, add an example like this one [1].

[1] https://github.com/open-education-hub/operating-systems/blob/7c49dbffecf3327f3ae46eaf842f2c09da3b773a/content/chapters/io/lecture/slides.md?plain=1#L55-L105

Reply:
I tried out a local demo. There's no animation there. These are just regular slides. What am I missing?

Reply:
There should be a transition when you scroll through them. Isn't it?

Reply:
No, it just adds content on the same slide when you scroll through them.